src/controller/proxy/local.go

```diff
@@ -135,6 +138,27 @@ func (l *localHelper) DeleteManifest(repo, ref string) {
 	}
 }
 
+// DeleteManifestQuiet delete manifest without existing check
+func (l *localHelper) DeleteManifestQuiet(ctx context.Context, repo, ref string) {
+	log.Infof("Remove tag from repo if it is exist, repo: %v ref: %v", repo, ref)
+	// convert tag to digest
+	bCtx := orm.Copy(ctx)
+	opt := &artifact.Option{}
+	art, err := l.artifactCtl.GetByReference(bCtx, repo, ref, opt)
+	if err != nil {
+		log.Debug("failed to get digest, skip to delete it")
+		return
+	}
+	if art == nil {
+		return
+	}
+	log.Infof("artifact does not exist in remote registry, going to remove it in local, artifact: %v@%v", repo, art.Digest)
+	if err := l.registry.DeleteManifestQuiet(repo, art.Digest); err != nil {
```

Comment:
if the purpose is to remove the artifact, I believe that this step is not needs since the GC will handle this part.